packages/validator-ajv8/src/validator.ts

```diff
@@ -245,22 +245,35 @@ export default class AJV8Validator<
     schema: RJSFSchema,
     formData?: T
   ): { errors?: Result[]; validationError?: Error } {
-    let validationError: Error | undefined = undefined;
+    let compilationError: Error | undefined = undefined;
+    let compiledValidator: ValidateFunction | undefined;
+    if (schema["$id"]) {
+      compiledValidator = this.ajv.getSchema(schema["$id"]);
+    }
     try {
-      this.ajv.validate(schema, formData);
+      if (compiledValidator === undefined) {
+        compiledValidator = this.ajv.compile(schema);
+      }
+      compiledValidator(formData);
```

Comment:
`validate` throws an error if you try to compile a schema that you've already compiled (based on $id).

This may also improve performance.

Reply:
Do we want to cache the compiled validator using a Map with the schema as a key? I'm not sure how much time it takes to compile it, but if it is a big schema it might save time on subsequent validations?

Reply:
I'd rather use the Ajv cache if possible, one less thing for us to get wrong. Though that might not work for conditional logic where we may create various subschemas that have no $id